### ISSUE TITLE
Issue with Pie Charts after version 4.1.1

### ISSUE THREAD

**Issue by dejongch** [highslide-software/highcharts.com#4100]
This is related to #4069 but I think there are more issues with the pie charts

Specifically I have a button that when pressed with either enable or disable all the pie slices depending on if all the slices are currently selected.

This jsfiddle (http://jsfiddle.net/hs9z6o2v/) has something similar to what I am doing and with it the legend items and the datalabels do not update properly. This works fine in 4.1.1.

On my application I also get more serious issues where in IE 11 the enable all button seems to completely freeze the system and it never comes back. In Chrome it freezes for awhile but does come back however the pie only has some slices drawn and the redraw function does not seem to see the changes done to the visible bools. I can't however get this to reproduce in a jsfiddle.

**Comment by dejongch** [comment]
Looking into it more I found that it was the fix to #4069 that is causing my newest issues. If I grab the code before that revision I do get the slow loads on very large pie graphs but I don't get the issues where the visibility of each slice not being set properly.

**Comment by dejongch** [comment]
More delving into this issue I found that there was a better way to set the visibility of pies with the series[0].points[x].setVisible. When I switch over to this some of my issues were fixed. When there are [few slices](http://jsfiddle.net/u6vg19dc/) this works fine. However, when there are a [huge number of slices](http://jsfiddle.net/xy4kauuw/1/) it takes minutes to update. Looking at it I think the issue lies in this peice of code:

```
if (point.legendItem) {
    if (chart.hasRendered) {
        series.updateTotals();
        chart.legend.clearItems();
        if (!doRedraw) {
            chart.legend.render();
        }
    }
    chart.legend.colorizeItem(point, vis);
}
```

Because I set the series to dirty so it won't redraw the whole graph every time it is still redrawing the whole legend each time if I am reading this right.
